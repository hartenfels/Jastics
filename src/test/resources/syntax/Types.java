import java.util.Set;

public class Types knows "music.rdf" {
  public static void genericTypes() {
    Set<«:MusicArtist»> s1 = do-query(":MusicArtist");
    Set<«:MusicArtist» ⊔ «:RadioStation»> s2 = do-query(":MusicArtist" ⊔ ":RadioStation");
    Set<«:RadioStation» ⊔ «:MusicArtist»> s3 = do-query(":MusicArtist" ⊔ ":RadioStation");
  }
}
