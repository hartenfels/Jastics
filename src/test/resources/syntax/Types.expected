import java.util.Set;

public class Types from "music.rdf" {
  public static void genericTypes() {
    Set<«:MusicArtist»> s1 = (query-for(":MusicArtist"));
    Set<«:MusicArtist» ⊔ «:RadioStation»> s2 = (query-for((":MusicArtist" ⊔ ":RadioStation")));
    Set<«:RadioStation» ⊔ «:MusicArtist»> s3 = (query-for((":MusicArtist" ⊔ ":RadioStation")));
  }

  public static void wildcardTypes() {
    Set<? extends «:MusicArtist»> s1 = (query-for(":MusicArtist"));
    Set<? extends «:MusicArtist» ⊔ «:RadioStation»> s2 = (query-for((":MusicArtist" ⊔ ":RadioStation")));
    Set<? extends «:RadioStation» ⊔ «:MusicArtist»> s3 = (query-for((":MusicArtist" ⊔ ":RadioStation")));
  }
}
